spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigDataLocationResolver.java

```diff
@@ -51,91 +51,52 @@ public Fabric8ConfigDataLocationResolver(DeferredLogFactory factory) {
 	protected void registerBeans(ConfigDataLocationResolverContext resolverContext, ConfigDataLocation location,
 			Profiles profiles, KubernetesConfigDataLocationResolver.PropertyHolder propertyHolder,
 			KubernetesNamespaceProvider namespaceProvider) {
-		KubernetesClientProperties properties = propertyHolder.kubernetesClientProperties();
+		KubernetesClientProperties kubernetesClientProperties = propertyHolder.kubernetesClientProperties();
 		ConfigMapConfigProperties configMapProperties = propertyHolder.configMapConfigProperties();
 		SecretsConfigProperties secretsProperties = propertyHolder.secretsProperties();
 
 		ConfigurableBootstrapContext bootstrapContext = resolverContext.getBootstrapContext();
-		Config config = config(properties);
-		bootstrapContext.registerIfAbsent(Config.class, InstanceSupplier.of(config));
-		bootstrapContext.addCloseListener(event -> event.getApplicationContext().getBeanFactory()
-				.registerSingleton("fabric8Config", event.getBootstrapContext().get(Config.class)));
-
-		KubernetesClient kubernetesClient = kubernetesClient(config);
-		bootstrapContext.registerIfAbsent(KubernetesClient.class,
-				BootstrapRegistry.InstanceSupplier.of(kubernetesClient));
-		bootstrapContext.addCloseListener(event -> event.getApplicationContext().getBeanFactory()
-				.registerSingleton("configKubernetesClient", event.getBootstrapContext().get(KubernetesClient.class)));
-
-		if (isRetryEnabled(configMapProperties, secretsProperties)) {
```

Comment:
this is a little involved. We used to have:

```
if (isRetryEnabled(configMapProperties, secretsProperties)) {
      registerRetryBeans(...)  
}

else {
    if (configMapProperties != null && configMapProperties.enabled()) {
            // register `Fabric8ConfigMapPropertySourceLocator `
    }

    if (secretsProperties != null && secretsProperties.enabled()) {
            // register `Fabric8SecretsPropertySourceLocator `
    }
}
```

The first question I had is why we register configmap retry bean (for example) if potentially, you can have  `spring.cloud.kubernetes.config.enabled=false`? Because in the `isRetryEnabled(configMapProperties, secretsProperties)`, we do not check for `spring.cloud.kubernetes.config.enabled`. 

As a counter example, in the non-configdata we have: 

```
@ConditionalOnKubernetesConfigEnabled
....
public @interface ConditionalOnKubernetesConfigRetryEnabled {

}
```

So in the bootstrap case, if we wanted retry beans to be present, you needed `@ConditionalOnKubernetesConfigEnabled`. Which to me, makes total sense.

---

And if the above is correct, then registering beans is simpler:


```
if (configMapProperties != null && configMapProperties.enabled()) {
			ConfigMapPropertySourceLocator configMapPropertySourceLocator = new Fabric8ConfigMapPropertySourceLocator(
				kubernetesClient, configMapProperties, namespaceProvider);
			if (isRetryEnabledForConfigMap(configMapProperties)) {
				configMapPropertySourceLocator = new ConfigDataRetryableConfigMapPropertySourceLocator(
					configMapPropertySourceLocator, configMapProperties);
			}

			registerSingle(bootstrapContext, ConfigMapPropertySourceLocator.class,
				configMapPropertySourceLocator, "configDataConfigMapPropertySourceLocator");
		}
```

We first compute the non-retry bean (only if `spring.cloud.kubernetes.config.enabled=true`, explicit or via `@Default` which is set to true), and if, retry is enabled, we would compute a retry bean. 

I am not exactly sure I am able to explain it properly and verbose enough... I surely hope so.